utilities/transactions/lock/range/range_tree/range_tree_lock_manager.cc

```diff
@@ -0,0 +1,572 @@
+    EndpointWithString start;
+    EndpointWithString end;
+    deserialize_endpoint(start_dbt, &start);
+    deserialize_endpoint(end_dbt, &end);
+
+    di_path.push_back({((PessimisticTransaction*)txnid)->GetID(),
+                       column_family_id, is_exclusive, std::move(start),
+                       std::move(end)});
+  };
+
+  request.start();
+
+  /*
+    If we are going to wait on the lock, we should set appropriate status in
+    the 'txn' object. This is done by the SetWaitingTxn() call below.
+    The API we are using are MariaDB's wait notification API, so the way this
+    is done is a bit convoluted.
+    In MyRocks, the wait details are visible in I_S.rocksdb_trx.
+  */
+  std::string key_str(start_key.data(), start_key.size());
+  struct st_wait_info {
+    PessimisticTransaction* txn;
+    uint32_t column_family_id;
+    std::string* key_ptr;
+    autovector<TransactionID> wait_ids;
+    bool done = false;
+
+    static void lock_wait_callback(void* cdata, TXNID /*waiter*/,
+                                   TXNID waitee) {
+      TEST_SYNC_POINT("RangeTreeLockManager::TryRangeLock:WaitingTxn");
+      auto self = (struct st_wait_info*)cdata;
+      // we know that the waiter is self->txn->GetID() (TODO: assert?)
+      if (!self->done) {
+        self->wait_ids.push_back(waitee);
+        self->txn->SetWaitingTxn(self->wait_ids, self->column_family_id,
+                                 self->key_ptr);
+        self->done = true;
+      }
+    }
+  } wait_info;
+
+  wait_info.txn = txn;
+  wait_info.column_family_id = column_family_id;
+  wait_info.key_ptr = &key_str;
+  wait_info.done = false;
+
+  const int r =
+      request.wait(wait_time_msec, killed_time_msec,
+                   nullptr,  // killed_callback
+                   st_wait_info::lock_wait_callback, (void*)&wait_info);
+
+  // Inform the txn that we are no longer waiting:
+  txn->ClearWaitingTxn();
+
+  request.destroy();
+  switch (r) {
+    case 0:
+      break; /* fall through */
+    case DB_LOCK_NOTGRANTED:
+      return Status::TimedOut(Status::SubCode::kLockTimeout);
+    case TOKUDB_OUT_OF_LOCKS:
+      return Status::Busy(Status::SubCode::kLockLimit);
+    case DB_LOCK_DEADLOCK: {
+      std::reverse(di_path.begin(), di_path.end());
+      dlock_buffer_.AddNewPath(
+          RangeDeadlockPath(di_path, request.get_start_time()));
+      return Status::Busy(Status::SubCode::kDeadlock);
+    }
+    default:
+      assert(0);
+      return Status::Busy(Status::SubCode::kLockLimit);
+  }
+
+  // Don't: save the acquired lock in txn->owned_locks.
+  // It is now responsibility of RangeTreeLockManager
+  //  RangeLockList* range_list=
+  //  ((RangeTreeLockTracker*)txn->tracked_locks_.get())->getOrCreateList();
+  //  range_list->append(column_family_id, &start_key_dbt, &end_key_dbt);
+
+  return Status::OK();
+}
+
+static void range_lock_mgr_release_lock_int(toku::locktree* lt,
```

Comment:
what's the meaning of `int`? Does it mean intention?

Reply:
"internal".  I see this function is called only from one place now, so I'll remove it.